src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -393,9 +393,7 @@ private List<Object> maybeExpandDirectories(
     }
 
     private static boolean hasDirectory(List<Object> originalValues) {
-      int n = originalValues.size();
-      for (int i = 0; i < n; ++i) {
```

Comment:
Was it written this way to reduce iterator garbage?

Reply:
I applied the IntelliJ suggestion after migrating the cast below without giving it much thought, but a JMH benchmark shows a noticeable difference.

Do you happen to have a rule of thumb for this? At least for arrays, I would expect identical performance, but probably not for generic lists?

Reply:
You're right it should be equivalent for arrays. For lists I would just assume that any existing `for (int i` style loops were chosen for performance, unless I know for sure that it's not a hot code path.